src/plugins/share/README.mdx

```diff
@@ -0,0 +1,243 @@
+        name: 'Demo list item (from share_example plugin)',
+      },
+    }
+  ],
+});
+```
+
+Now the "Demo list item" will appear under the "Share" menu in Discover and
+Dashboard applications.
+
+
+## Locators
+
+*Locators* are entities which given some *parameters* can navigate Kibana user
+to some deep link in Kibana or return back a full formatted URL.
+
+Kibana apps create locators and expose them from their plugin contracts. Then
+other plugins can use those locators to navigate deeply into Kibana applications.
+
+Locators work, both, on the server and browser. However, it is the responsibility
+of the app, which provides the locator, to register and expose the same locator
+on the server and browser.
+
+### Creating a locator for your Kibana app
+
+A Kibana app can create one or more *locators*, which given an object of *parameters*
+knows how to specify return a *Kibana location* 3-tuple. The 3-tuple consists of:
+(1) the app ID; (2) a path within the app; (3) a serializable *location state* object.
+The Kibana location 3-tuple is used to navigate the Kibana app in SPA (single page app)
+way&mdash;without reloading the page.
+
+```ts
+import type { SerializableRecord } from '@kbn/utility-types';
+import { LocatorDefinition, LocatorPublic } from 'src/plugins/share/public';
+
+export interface HelloLocatorParams extends SerializableRecord {
+  name: string;
+}
+
+export type HelloLocator = LocatorPublic<HelloLocatorParams>;
+
+export class HelloLocatorDefinition implements LocatorDefinition<HelloLocatorParams> {
+  public readonly id = 'HELLO_LOCATOR';
+
+  public readonly getLocation = async ({ name }: HelloLocatorParams) => {
+    return {
+      app: 'sampleAppId',
+      path: `/hello?name=${encodeURIComponent(name)}`,
+      state: {},
+    };
+  };
+}
+```
+
+Once your locator has been defined, you need to register it in the central locator
+registry with the `share` plugin:
+
+```ts
+const locator = plugins.share.url.locators.create(new HelloLocatorDefinition());
+```
+
+You can then return the `locator` reference from your plugin contract fro consumption
+by other plugins:
+
+```ts
+class MyPlugin {
+  setup(core, plugins) {
+    const locator = plugins.share.url.locators.create(new HelloLocatorDefinition());
+
+    return {
+      locator,
+    };
+  }
+}
+```
+
+### Using locator of another app
+
+First you need to get access of the locator you want to use. Best way to do it is
+by acquiring it from the plugin contract of the app you are interested to navigate
+to. For example, you can get hold of the Dashboard app locator from the `dashboard`
+plugin contract:
+
+```ts
+class MyPlugin {
+  setup(core, plugins) {
+    const dashboardLocator = plugins.dashboard.locator;
+  }
+}
+```
+
+Alternatively, you can also acquire any locator from the central registry in
+the `share` plugin using the ID of the locator. This is useful, if you run into
+circular dependency issues, however, it is better to acquire the locator directly
```

Comment:
i am not sure i would mention this, we don't want ppl to use locators from the registry imo unless they have a usecase where they don't know the locator they'll be using at compile time.